File: android/core/domain/src/main/java/com/enterprise/core/domain/usecases/LoggerUseCase.kt
package com.enterprise.core.domain.usecases

import com.enterprise.core.domain.interfaces.AnalyticsTracker
import javax.inject.Inject

/** AI agent logs intent resolution decisions (local vs. cloud LLM). */
class LoggerUseCase @Inject constructor(
    private val analyticsTracker: AnalyticsTracker
) {
    fun logLocalResolution(intent: String, confidence: Double) {
        analyticsTracker.trackEvent("agent_intent_resolved", mapOf(
            "intent" to intent,
            "source" to "local_llm",
            "confidence" to confidence.toString()
        ))
    }

    fun logCloudFallback(intent: String, reason: String) {
        analyticsTracker.trackEvent("agent_intent_resolved", mapOf(
            "intent" to intent,
            "source" to "cloud_llm",
            "fallback_reason" to reason
        ))
    }
}

File: ios/EnterpriseMobileApp/Sources/Core/Domain/UseCases/LoggerUseCase.swift
import Foundation

/// Use case for the AI Agent to log intent resolution decisions.
public struct LoggerUseCase: Sendable {
    private let analyticsTracker: AnalyticsTracker
    private let crashReporter: CrashReporter

    public init(analyticsTracker: AnalyticsTracker, crashReporter: CrashReporter) {
        self.analyticsTracker = analyticsTracker
        self.crashReporter = crashReporter
    }

    public func logLocalResolution(intent: String, confidence: Double) {
        analyticsTracker.trackEvent("ai_intent_resolved", properties: [
            "intent": intent,
            "source": "local_llm",
            "confidence": confidence
        ])
        crashReporter.logBreadcrumb("AI resolved '\(intent)' locally (confidence: \(confidence))")
    }

    public func logCloudFallback(intent: String, reason: String) {
        analyticsTracker.trackEvent("ai_intent_fallback", properties: [
            "intent": intent,
            "source": "cloud_llm",
            "reason": reason
        ])
        crashReporter.logBreadcrumb("AI fell back to cloud for '\(intent)': \(reason)")
    }
}
